autoshop/core/views.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import unicode_literals
+
+from django.contrib.auth import authenticate, login
+from django.contrib.auth.models import User
+from django.contrib.auth import logout
+from django.db import IntegrityError
+from django.shortcuts import render, redirect
+from .forms import SignupForm, SigninForm
+
+
+def home(request):
+    """
+    This view redirects users to the main page of autoshop website.
+    """
+    return render(request, 'home.html')
+
+
+def register(request):
+    """
+    This view redirects users to the signup page. If user is submitting
+    form (POST), user registration takes place. If user is visiting
+    page, user is directed to page containing reqistration form.
+
+    """
+    if request.method == 'POST':
+        form = SignupForm(request.POST)
+        if form.is_valid():
+            try:
+                User.objects.create_user(
+                    first_name=form.data['first_name'],
+                    last_name=form.data['last_name'],
+                    email=form.data['email'],
+                    password=form.data['password'],
+                    username=form.data['email'],
+                )
+                return redirect('/')
+            except IntegrityError:
```

Comment:
No. I tried to create the user with get_or_create to avoid exception but that causes login problems.